substrate/frame/multisig/src/lib.rs

```diff
@@ -539,6 +679,18 @@ impl<T: Config> Pallet<T> {
 			let timepoint = maybe_timepoint.ok_or(Error::<T>::NoTimepoint)?;
 			ensure!(m.when == timepoint, Error::<T>::WrongTimepoint);
 
+			let maybe_expiry = <MultisigExpiries<T>>::get(&id, call_hash);
+			// Ensure that the mutlisig did not expire.
+			match maybe_expiry {
+				Some(expiry) if expiry < <frame_system::Pallet<T>>::block_number() => {
+					// If the multisig is expired we will take the chance to remove it now so that
+					// the multisig creator does not have to make a separate call to clean it up.
+					Self::remove_multisig(&id, call_hash, &m.depositor, m.deposit, maybe_expiry);
+					return Err(Error::<T>::MultisigExpired.into())
+				},
+				_ => {},
+			};
```

Comment:
I think it would be cleaner to replace this match with an `if let Some(expiry)` statement

Reply:
I do not have a strong opinion on this, but I wanted to do it this way because for some reason I thought it looked nicer than having a nested if statement 🤷‍♂️

Reply:
After looking at this again, I also think using an if statement looks better.